_rules/SC2-4-6-descriptive-headings.md

```diff
@@ -22,18 +23,24 @@ authors:
 
 ### Applicability
 
-This rule applies to any element with the [semantic role](#semantic-role) of heading that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree).
+This rule applies to any element with the [semantic role](#semantic-role) of `heading` that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree), and that either has [visible](#visible) [content](https://www.w3.org/TR/WCAG21/#dfn-content), or has an [accessible name](#accessible-name) that does not only consist of [Unicode separator characters](https://www.unicode.org/versions/Unicode11.0.0/ch04.pdf#G134153).
```

Comment:
Sorry, but the definition of "content" is subjective and so can't be used in applicability. I don't think we can create an objective definition of what WCAG considers "content". I'm not sure I have a good solution for this. The best thing I've come up with in thinking about this for 15 min is to just look at any visible heading, or any heading with an accessible name. A heading might be visible without it having content, for instance if it was just a line or something, but the expectations would still be valid.

Reply:
I think this suggestion would work. Simply look at all visible headings.